applications/StructuralMechanicsApplication/custom_response_functions/adjoint_conditions/adjoint_semi_analytic_base_condition.h

```diff
@@ -91,224 +94,231 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION)  AdjointSemiAnalyticBaseCondi
+        mPrimalCondition.CalculateFirstDerivativesContributions(rLeftHandSideMatrix,
 							rRightHandSideVector,
 							rCurrentProcessInfo);
     }
 
     void CalculateFirstDerivativesLHS(MatrixType& rLeftHandSideMatrix,
 					      ProcessInfo& rCurrentProcessInfo) override
     {
-        mpPrimalCondition->CalculateFirstDerivativesLHS(rLeftHandSideMatrix,
+        mPrimalCondition.CalculateFirstDerivativesLHS(rLeftHandSideMatrix,
 					    rCurrentProcessInfo);
     }
 
     void CalculateFirstDerivativesRHS(VectorType& rRightHandSideVector,
 					      ProcessInfo& rCurrentProcessInfo) override
     {
-        mpPrimalCondition->CalculateFirstDerivativesRHS(rRightHandSideVector,
+        mPrimalCondition.CalculateFirstDerivativesRHS(rRightHandSideVector,
 					      rCurrentProcessInfo);
     }
 
     void CalculateSecondDerivativesContributions(MatrixType& rLeftHandSideMatrix,
 							 VectorType& rRightHandSideVector,
 							 ProcessInfo& rCurrentProcessInfo) override
     {
-        mpPrimalCondition->CalculateSecondDerivativesContributions(rLeftHandSideMatrix,
+        mPrimalCondition.CalculateSecondDerivativesContributions(rLeftHandSideMatrix,
 							 rRightHandSideVector,
 							 rCurrentProcessInfo);
     }
 
     void CalculateSecondDerivativesLHS(MatrixType& rLeftHandSideMatrix,
 					       ProcessInfo& rCurrentProcessInfo) override
     {
-        mpPrimalCondition->CalculateSecondDerivativesLHS(rLeftHandSideMatrix,
+        mPrimalCondition.CalculateSecondDerivativesLHS(rLeftHandSideMatrix,
 					       rCurrentProcessInfo);
     }
 
     void CalculateSecondDerivativesRHS(VectorType& rRightHandSideVector,
 					       ProcessInfo& rCurrentProcessInfo) override
     {
-        mpPrimalCondition->CalculateSecondDerivativesRHS(rRightHandSideVector,
+        mPrimalCondition.CalculateSecondDerivativesRHS(rRightHandSideVector,
 					       rCurrentProcessInfo);
     }
 
     void CalculateMassMatrix(MatrixType& rMassMatrix, ProcessInfo& rCurrentProcessInfo) override
     {
-        mpPrimalCondition->CalculateMassMatrix(rMassMatrix, rCurrentProcessInfo);
+        mPrimalCondition.CalculateMassMatrix(rMassMatrix, rCurrentProcessInfo);
     }
 
     void CalculateDampingMatrix(MatrixType& rDampingMatrix, ProcessInfo& rCurrentProcessInfo) override
     {
-        mpPrimalCondition->CalculateDampingMatrix(rDampingMatrix, rCurrentProcessInfo);
+        mPrimalCondition.CalculateDampingMatrix(rDampingMatrix, rCurrentProcessInfo);
     }
 
-    void Calculate(const Variable<double >& rVariable,
+    virtual void Calculate(const Variable<double >& rVariable,
 			   double& Output,
 			   const ProcessInfo& rCurrentProcessInfo) override
     {
         KRATOS_ERROR << "Calculate of the adjoint base condition is called!" << std::endl;
     }
 
-    void Calculate(const Variable< array_1d<double,3> >& rVariable,
+    virtual void Calculate(const Variable< array_1d<double,3> >& rVariable,
```

Comment:
why do you add `virtual`? => not needed if it is an override

Reply:
because it gets again overridden in the point load condition.

Reply:
you only need virtual once, in the baseclass
All derived classes dont need it

Reply:
You were right, i reverted it